protocol/parser/parser.go

```diff
@@ -281,7 +302,11 @@ func (p *ParsedInput) GetBlockHashes() ([]string, error) {
 }
 
 func getMapForParse(rpcInput RPCInput) map[string]interface{} {
-	return map[string]interface{}{"params": rpcInput.GetParams(), "result": rpcInput.GetResult()}
+	var result map[string]interface{}
+	if rpcInput.GetResult() != nil {
```

Comment:
no need to call getResult twice, save the value and then unmarshal from it

Reply:
Fixed in 9d88cd7ab3cdde1f27be72c2cf857f06cfb5700e